src/features/search/pages/modals/VenueModal/useVenueModal.native.test.tsx

```diff
@@ -102,17 +108,64 @@ describe('useVenueModal', () => {
     })
     expect(dismissMyModal).toHaveBeenCalledWith()
     expect(mockStateDispatch).toHaveBeenCalledWith({
-      type: 'SET_LOCATION_VENUE',
-      payload: venue,
+      type: 'SET_STATE',
+      payload: {
+        ...initialSearchState,
+        locationFilter: { locationType: LocationType.VENUE, venue: venue },
+        view: SearchView.Results,
+      },
     })
   })
   it('when nothing is to be search then search cannot be done', async () => {
-    const { result } = renderHook(useVenueModal, { initialProps: dismissMyModal })
+    const { result } = renderHook(useVenueModal, { initialProps: { dismissModal: dismissMyModal } })
+
+    await act(async () => {
+      result.current.doApplySearch()
+    })
+    expect(doAfterSearch).not.toHaveBeenCalledWith()
+    expect(dismissMyModal).toHaveBeenCalledWith()
+    expect(mockStateDispatch).not.toHaveBeenCalledWith()
+  })
+  it('when nothing is to be search then search cannot be done and no calls is made', async () => {
+    const { result } = renderHook(
+      ({ dismissModal, doAfterSearch }) => useVenueModal({ dismissModal, doAfterSearch }),
```

Comment:
pourrait-on simplifier comme ça ?

```suggestion
      useVenueModal,
```

Reply:
Yep